android/java/org/chromium/chrome/browser/crypto_wallet/util/ImageLoader.java

```diff
@@ -0,0 +1,84 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.chromium.chrome.browser.crypto_wallet.util;
+
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.graphics.drawable.Drawable;
+import android.graphics.drawable.VectorDrawable;
+import android.text.TextUtils;
+import android.util.Base64;
+import android.widget.ImageView;
+
+import com.bumptech.glide.Glide;
+import com.bumptech.glide.Priority;
+import com.bumptech.glide.RequestBuilder;
+import com.bumptech.glide.load.engine.DiskCacheStrategy;
+import com.bumptech.glide.load.resource.gif.GifDrawable;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class ImageLoader {
+    private static final List<String> ANIMATED_lIST = Arrays.asList(".gif");
+
+    public static void loadNft(String url, ImageView imageView, Context context,
+                               boolean isCircular) {
+        if (isAnimated(url)) {
+            loadGif(url, imageView, context, isCircular);
+        }else {
+            loadImg(url, imageView, context, isCircular);
+        }
+    }
+
+    public static void loadImg(String url, ImageView imageView, Context context,
+                               boolean isCircular) {
+        RequestBuilder<Drawable> request = Glide.with(context).load(url);
+        if (isCircular) {
+            request = request.circleCrop();
+        }
+        request.priority(Priority.IMMEDIATE)
+                .diskCacheStrategy(DiskCacheStrategy.ALL)
+                .into(imageView);
+    }
+
+    public static void loadGif(String url, ImageView imageView, Context context,
+                               boolean isCircular) {
+        RequestBuilder<GifDrawable> request = Glide.with(context).asGif().load(url);
+        if (isCircular) {
+            request = request.circleCrop();
+        }
+        request.circleCrop()
+                .priority(Priority.IMMEDIATE)
+                .diskCacheStrategy(DiskCacheStrategy.ALL)
+                .into(imageView);
+    }
+
+    public static boolean isSupported(String url){
+        return !isSvg(url);
+    }
+
+    public static boolean isAnimated(String url) {
+        if (isValidImgUrl(url)) return false;
+        for (String extension : ANIMATED_lIST) {
+            if (url.endsWith(extension)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public static boolean isSvg(String url) {
+        if (isValidImgUrl(url)) return false;
+        return url.startsWith("data:image/svg") || url.endsWith(".svg");
+    }
+
+
+    private static boolean isValidImgUrl(String url) {
```

Comment:
Is this logic correct?

Reply:
This is just an internal method for small checks. Right now we don't have any specific requirement so not adding anything. though we might update it later if required.

Reply:
Okay, maybe you meant 
`!TextUtils.isEmpty(url);` (With the excalmation mark at the beginning). 

Right now if the url is null the method return `true`.

Reply:
Yes, thanks.

Reply:
done